src/NuGet.Core/NuGet.Protocol.Core.v3/Resources/AutoCompleteResourceV3.cs

```diff
@@ -43,7 +44,7 @@ public AutoCompleteResourceV3(HttpSource client, ServiceIndexResourceV3 serviceI
             // Construct the query
             var queryUrl = new UriBuilder(searchUrl.AbsoluteUri);
             var queryString =
-                "q=" + packageIdPrefix + "&includePrerelease=" + includePrerelease;
+                "q=" + WebUtility.UrlEncode(packageIdPrefix) + "&includePrerelease=" + includePrerelease;
```

Comment:
we should probably .ToString with a format for includePrelease to make it explicit